sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/DefaultOfferRequirementProvider.java

```diff
@@ -51,11 +52,116 @@ public OfferRequirement getNewOfferRequirement(PodInstance podInstance, Collecti
         return OfferRequirement.create(
                 podInstance.getPod().getType(),
                 podInstance.getIndex(),
-                getNewTaskInfos(podInstance, tasksToLaunch, serviceName, targetConfigurationId),
-                Optional.of(getExecutor(podInstance, serviceName, targetConfigurationId)),
+                getNewTaskRequirements(podInstance, tasksToLaunch, serviceName, targetConfigurationId),
+                getExecutorRequirement(podInstance, serviceName, targetConfigurationId),
                 podInstance.getPod().getPlacementRule());
     }
 
+    private Collection<TaskRequirement> getNewTaskRequirements(
+            PodInstance podInstance,
+            Collection<String> tasksToLaunch,
+            String serviceName,
+            UUID targetConfigurationId) throws InvalidRequirementException{
+        LOGGER.info("Getting new TaskRequirements for tasks: {}", tasksToLaunch);
+
+        ArrayList<String> usedResourceSets = new ArrayList<>();
+        List<TaskRequirement> taskRequirements = new ArrayList<>();
+
+        // Generating TaskInfos to launch.
+        for (TaskSpec taskSpec : podInstance.getPod().getTasks()) {
+            if (!tasksToLaunch.contains(taskSpec.getName())) {
+                continue;
+            }
+
+            if (!usedResourceSets.contains(taskSpec.getResourceSet().getId())) {
+                LOGGER.info("Generating taskInfo to launch for: {}, with resource set: {}",
+                        taskSpec.getName(), taskSpec.getResourceSet().getId());
+                usedResourceSets.add(taskSpec.getResourceSet().getId());
+                taskRequirements.add(getNewTaskRequirement(
+                        podInstance, taskSpec, serviceName, targetConfigurationId, false));
+            }
```

Comment:
The DefaultOfferRequirementProvider tests already check for this behavior, I believe -- I just adapted it to return TaskRequirements instead of TaskInfos, but it was already existing.